### PR TITLE
Refactor encryption dir in config

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description
 Refactor encryption dir in config

### Changed

- Remove dict and legacy entry. Replace with string
- Decouple from file system path when querying PDC
- Add flow cell specific search pattern
- Simplify extracting file path from DCMS output

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b flat_encrypt_dir -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by henrikstranneheim** [comment]
Linked to: https://github.com/Clinical-Genomics/servers/pull/1124

File: tests/conftest.py [hunk 1]
```diff
@@ -2215,10 +2218,7 @@ def context_config(
         "madeline_exe": "echo",
         "pon_path": str(cg_dir),
         "backup": {
-            "encrypt_dir": {
-                "current": str(current_encryption_dir),
-                "legacy": str(legacy_encryption_dir),
-            },
+            "encrypt_dir": str(current_encryption_dir),
             "root": {"hiseqx": "flowcells/hiseqx", "hiseqga": "RUNS/", "novaseq": "runs/"},
```
Comment (ChrOertlin) [inline tests/conftest.py:2222]:
```suggestion
            
```

File: cg/meta/backup/backup.py [hunk 2]
```diff
@@ -294,31 +292,29 @@ def get_archived_flow_cell_path(self, query: list) -> Path:
...
-        for dir in self.encrypt_dir.values():
-            re_archived_flow_cell_path: re.Pattern = re.compile(dir + ".+?(?=\s)")
-            arhived_flow_cell: Optional[re.Match] = re.search(
-                re_archived_flow_cell_path, flow_cell_query
-            )
-            if arhived_flow_cell:
-                archived_flow_cell_path: Path = Path(arhived_flow_cell.group())
-                LOG.info(f"Flow cell found: {archived_flow_cell_path}")
-                return archived_flow_cell_path
+        re_archived_flow_cell_path: re.Pattern = re.compile(self.encrypt_dir + ".+?(?=\s)")
```
Comment (ChrOertlin) [inline cg/meta/backup/backup.py:295]:
```suggestion
        re_archived_flow_cell_path: re.Pattern = re.compile(self.encrypt_dir + ".+?(?=\s)")
```
should we add some info what this pattern matches?

File: cg/meta/backup/backup.py [hunk 3]
```diff
@@ -294,31 +292,29 @@ def get_archived_flow_cell_path(self, query: list) -> Path:
...
+        arhived_flow_cell: Optional[re.Match] = re.search(
```
Comment (ChrOertlin) [inline cg/meta/backup/backup.py:296]:
```suggestion
        archived_flow_cell: Optional[re.Match] = re.search(
```

File: cg/meta/backup/backup.py [hunk 4]
```diff
@@ -294,31 +292,29 @@ def get_archived_flow_cell_path(self, query: list) -> Path:
...
+            re_archived_flow_cell_path, flow_cell_query
+        )
+        if arhived_flow_cell:
```
Comment (ChrOertlin) [inline cg/meta/backup/backup.py:299]:
```suggestion
        if archived_flow_cell:
```

File: cg/meta/backup/backup.py [hunk 5]
```diff
@@ -294,31 +292,29 @@ def get_archived_flow_cell_path(self, query: list) -> Path:
...
+            archived_flow_cell_path: Path = Path(arhived_flow_cell.group())
```
Comment (ChrOertlin) [inline cg/meta/backup/backup.py:300]:
```suggestion
            archived_flow_cell_path: Path = Path(archived_flow_cell.group())
```

**Review by ChrOertlin — APPROVED** [review]
Looks good! Nice that we can simplify this

**Comment by henrikstranneheim** [comment]
### Fetching flow cell 🚫 

```
θ60° 5d [hiseq.clinical@hasta:~] [S_base] 22s 1 $ cg backup  fetch-flow-cell -d -f HJFW5DSXX
Running command /bin/dsmc q archive /home/proj/stage/encrypt/*HJFW5DSXX*
Call ['/bin/dsmc', 'q', 'archive', '/home/proj/stage/encrypt/*HJFW5DSXX*'] exit with a non zero exit code
ANS1092W No files matching search criteria were found
No archived files found for pdc query ['/home/proj/stage/encrypt/*HJFW5DSXX*'], testing legacy directory
PDC query failed: No archived files found for pdc queries ['/home/proj/stage/encrypt/*HJFW5DSXX*']
Traceback (most recent call last):
  File "/home/proj/stage/bin/miniconda3/envs/S_main/bin/cg", line 8, in <module>
    sys.exit(base())
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/cli/backup.py", line 63, in fetch_flow_cell
    retrieval_time: Optional[float] = backup_api.fetch_flow_cell(flow_cell=flow_cell)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/meta/backup/backup.py", line 93, in fetch_flow_cell
    raise error
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/meta/backup/backup.py", line 89, in fetch_flow_cell
    pdc_flow_cell_query: List[str] = self.query_pdc_for_flow_cell(flow_cell.name)
  File "/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/cg/meta/backup/backup.py", line 272, in query_pdc_for_flow_cell
    message=f"No archived files found for pdc queries {search_patterns}"
cg.exc.PdcNoFilesMatchingSearchError: No archived files found for pdc queries ['/home/proj/stage/encrypt/*HJFW5DSXX*']
θ67° 5d [hiseq.clinical@hasta:~] [S_base] 23s 1 $ /bin/dsmc q archive /home/hiseq.clinical/ENCRYPT/*HJFW5DSXX*
IBM Tivoli Storage Manager
Command Line Backup-Archive Client Interface
  Client Version 7, Release 1, Level 4.0
  Client date/time: 09/22/2023 10:22:17
(c) Copyright by IBM Corporation and other(s) 1990, 2015. All Rights Reserved.

Node Name: HASTA.SCILIFELAB.SE_CLINICAL
Session established with server BLACKHOLE: Linux/x86_64
  Server Version 8, Release 1, Level 9.300
  Server date/time: 09/22/2023 10:22:17  Last access: 09/22/2023 10:17:49

Accessing as node: SLLCLINICAL
             Size  Archive Date - Time    File - Expires on - Description
             ----  -------------------    -------------------------------
           607  B  04/07/2019 07:30:15    /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.key.gpg Never Archive Date: 04/07/2019
 1,244,997,334 KB  04/07/2019 04:00:05    /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg Never Archive Date: 04/07/2019
```
Seems like the query return post filtering  is coupled to where it it was archived from. The flow cell id is unique so that should be enough. @Vince-janv @karlnyr Do you know if we need the legacy path downstream?

**Comment by henrikstranneheim** [comment]
This seem to be an option:
```
θd [hiseq.clinical@hasta:~] [S_base] 4s 8 $ /bin/dsmc q archive */*HJFW5DSXX*
IBM Tivoli Storage Manager
Command Line Backup-Archive Client Interface
  Client Version 7, Release 1, Level 4.0
  Client date/time: 09/22/2023 10:44:57
(c) Copyright by IBM Corporation and other(s) 1990, 2015. All Rights Reserved.

Node Name: HASTA.SCILIFELAB.SE_CLINICAL
Session established with server BLACKHOLE: Linux/x86_64
  Server Version 8, Release 1, Level 9.300
  Server date/time: 09/22/2023 10:44:57  Last access: 09/22/2023 10:44:44

Accessing as node: SLLCLINICAL
             Size  Archive Date - Time    File - Expires on - Description
             ----  -------------------    -------------------------------
           607  B  04/07/2019 07:30:15    /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.key.gpg Never Archive Date: 04/07/2019
 1,244,997,334 KB  04/07/2019 04:00:05    /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg Never Archive Date: 04/07/2019
θ63° 5d [hiseq.clinical@hasta:~] [S_base] 9s $ /bin/dsmc q archive */*H7D68ADXX*
IBM Tivoli Storage Manager
Command Line Backup-Archive Client Interface
  Client Version 7, Release 1, Level 4.0
  Client date/time: 09/22/2023 10:46:09
(c) Copyright by IBM Corporation and other(s) 1990, 2015. All Rights Reserved.

Node Name: HASTA.SCILIFELAB.SE_CLINICAL
Session established with server BLACKHOLE: Linux/x86_64
  Server Version 8, Release 1, Level 9.300
  Server date/time: 09/22/2023 10:46:09  Last access: 09/22/2023 10:44:57

Accessing as node: SLLCLINICAL
             Size  Archive Date - Time    File - Expires on - Description
             ----  -------------------    -------------------------------
           607  B  02/10/2017 12:37:18    /home/hiseq.clinical/ENCRYPT/131029_D00134_0031_AH7D68ADXX.key.gpg Never Archive Date: 02/10/2017
46,324,877,738  B  02/10/2017 12:30:02    /home/hiseq.clinical/ENCRYPT/131029_D00134_0031_AH7D68ADXX.tar.gz.gpg Never Archive Date: 02/10/2017
θ62° 5d [hiseq.clinical@hasta:~] [S_base] 4s $ /bin/dsmc q archive */*H9ABBADXX*
IBM Tivoli Storage Manager
Command Line Backup-Archive Client Interface
  Client Version 7, Release 1, Level 4.0
  Client date/time: 09/22/2023 10:47:20
(c) Copyright by IBM Corporation and other(s) 1990, 2015. All Rights Reserved.

Node Name: HASTA.SCILIFELAB.SE_CLINICAL
Session established with server BLACKHOLE: Linux/x86_64
  Server Version 8, Release 1, Level 9.300
  Server date/time: 09/22/2023 10:47:21  Last access: 09/22/2023 10:46:09

Accessing as node: SLLCLINICAL
             Size  Archive Date - Time    File - Expires on - Description
             ----  -------------------    -------------------------------
           607  B  11/28/2016 11:36:14    /home/hiseq.clinical/ENCRYPT/140526_D00410_0072_BH9ABBADXX.key.gpg Never Archive Date: 11/28/2016
42,189,589,091  B  11/28/2016 11:30:03    /home/hiseq.clinical/ENCRYPT/140526_D00410_0072_BH9ABBADXX.tar.gz.gpg Never Archive Date: 11/28/2016
```

**Comment by henrikstranneheim** [comment]
### fetch flow cell dry-run ✅ 
```
67° 5d [hiseq.clinical@hasta:~] [S_base] 21s 1 $ cg backup  fetch-flow-cell -d -f HJFW5DSXX
Running command /bin/dsmc q archive */*HJFW5DSXX*
Found archived files for pdc query: */*HJFW5DSXX*
Encryption key found: 190329_A00689_0018_AHJFW5DSXX.key.gpg
Flow cell found: 190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg
```

**Comment by Vince-janv** [comment]
The only thing I can see that's problematic with `*/*HJFW5DSXX*` is that it will catch any file with the flow cell in the path ie any future spring file we send to PDC. Maybe we can include the gpg-suffix for a bit more specificity?

File: tests/conftest.py (hunk 1 above)
Comment (henrikstranneheim) [reply tests/conftest.py]: Done

File: cg/meta/backup/backup.py (hunk 2 above)
Comment (henrikstranneheim) [reply cg/meta/backup/backup.py]: Removed regexp and simplified using split instead -  fixed

File: cg/meta/backup/backup.py (hunk 4 above)
Comment (henrikstranneheim) [reply cg/meta/backup/backup.py]: Fixed

File: cg/meta/backup/backup.py (hunk 5 above)
Comment (henrikstranneheim) [reply cg/meta/backup/backup.py]: Fixed

File: cg/meta/backup/backup.py (hunk 3 above)
Comment (henrikstranneheim) [reply cg/meta/backup/backup.py]: fixed

**Comment by henrikstranneheim** [comment]
> The only thing I can see that's problematic with `*/*HJFW5DSXX*` is that it will catch any file with the flow cell in the path ie any future spring file we send to PDC. Maybe we can include the gpg-suffix for a bit more specificity?

Good suggestion - fixed

**Comment by henrikstranneheim** [comment]
Redid dry-run ✅ 
```
 [hiseq.clinical@hasta:~] [S_base] $ cg backup  fetch-flow-cell -d -f HJFW5DSXX
Rnning command /bin/dsmc q archive */*HJFW5DSXX*.gpg
Found archived files for PDC query: */*HJFW5DSXX*.gpg
Encryption key found: 190329_A00689_0018_AHJFW5DSXX.key.gpg
Flow cell found: 190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg
```

**Review by diitaz93 — APPROVED** [review]
Looks good 👍

**Comment by henrikstranneheim** [comment]
Retrieving flow cell and key using this code works, but errors on Selinux error related to retrieval of file from old file system. Retrieving file first time seems to work:
```
θ61° 5d [hiseq.clinical@hasta:~] [S_base] 6s 12 $ /bin/dsmc retrieve -replace=yes /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.key.gpg /home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.key.gpg
IBM Tivoli Storage Manager
Command Line Backup-Archive Client Interface
  Client Version 7, Release 1, Level 4.0
  Client date/time: 09/25/2023 08:43:07
(c) Copyright by IBM Corporation and other(s) 1990, 2015. All Rights Reserved.

Retrieve function invoked.

Node Name: HASTA.SCILIFELAB.SE_CLINICAL
Session established with server BLACKHOLE: Linux/x86_64
  Server Version 8, Release 1, Level 9.300
  Server date/time: 09/25/2023 08:43:07  Last access: 09/25/2023 08:42:45

Accessing as node: SLLCLINICAL
ANS1590W  I/O error writing file attribute: security.selinux for: /home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.key.gpg. errno = 1, Operation not permitted
Retrieving             607 /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.key.gpg --> /home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.key.gpg [Done]

Retrieve processing finished.

Total number of objects retrieved:            1
Total number of objects failed:               0
Total number of bytes transferred:          736  B
Data transfer time:                        0.00 sec
Network data transfer rate:          179,687.50 KB/sec
Aggregate data transfer rate:              0.01 KB/sec
Elapsed processing time:               00:00:45

```

```
5d [hiseq.clinical@hasta:~] [S_base] 23s $ cg backup  fetch-flow-cell -f HJFW5DSXX
HJFW5DSXX: retrieving from PDC
Running command /bin/dsmc q archive */*HJFW5DSXX*.gpg
Found archived files for PDC query: */*HJFW5DSXX*.gpg
Encryption key found: /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.key.gpg
Flow cell found: /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg
Running command /bin/dsmc retrieve -replace=yes /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.key.gpg /home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.key.gpg
Call ['/bin/dsmc', 'retrieve', '-replace=yes', '/home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.key.gpg', '/home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.key.gpg'] exit with a non zero exit code

WARNING for retrieval of encryption key of flow cell HJFW5DSXX, please check dsmerror.log
Running command /bin/dsmc retrieve -replace=yes /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg /home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg
```

```
63° 5d [hiseq.clinical@hasta:~] [S_base] $ ls -lh /home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.*
-rw-rwxr-- 1 hiseq.clinical clinical  607 Apr  6  2019 /home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.key.gpg
-rw-r--r-- 1 hiseq.clinical clinical 446G Sep 25 10:19 /home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg
θ65° 5d [hiseq.clinical@hasta:~] [S_base] $ ls -lh /home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.*
-rw-rwxr-- 1 hiseq.clinical clinical  607 Apr  6  2019 /home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.key.gpg
-rw-r--r-- 1 hiseq.clinical clinical 498G Sep 25 10:25 /home/proj/stage/flow_cells/190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg
```

@karlnyr @Vince-janv Do you approve merging this as the error is not related to this code?

**Comment by Vince-janv** [comment]
> Do you approve merging this as the error is not related to this code?

@henrikstranneheim Do you know if what the exit code is for the dsmc command? I think this might be happening in prod as well and related to #2472.

File: cg/meta/backup/backup.py [hunk 6]
```diff
@@ -294,31 +288,28 @@ def get_archived_flow_cell_path(self, query: list) -> Path:
...
-                return archived_flow_cell_path
+        file_colum: str = flow_cell_query.split()[4]
+        archived_flow_cell = Path(file_colum.split(sep="/")[-1])
+        if archived_flow_cell:
+            LOG.info(f"Flow cell found: {archived_flow_cell}")
+            return archived_flow_cell
 
-    def get_archived_encryption_key_path(self, query: list) -> Path:
+    @classmethod
+    def get_archived_encryption_key_path(cls, query: list) -> Path:
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply cg/meta/backup/backup.py]: It's defined locally so should be easy to refactor 👍 . Fixed

[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply cg/meta/backup/backup.py]: Fixed

File: cg/meta/backup/backup.py [hunk 7]
```diff
@@ -284,7 +277,8 @@ def retrieve_archived_file(self, archived_file: Path, run_dir: Path) -> None:
             file_path=str(archived_file), target_path=str(retrieved_file)
         )
 
-    def get_archived_flow_cell_path(self, query: list) -> Path:
+    @classmethod
+    def get_archived_flow_cell_path(cls, query: list) -> Path:
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply cg/meta/backup/backup.py]: Fixed

[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply cg/meta/backup/backup.py]: Fixed

**Comment by henrikstranneheim** [comment]
Test pass
```
θ63° 5d [hiseq.clinical@hasta:~] [S_base] 127 $ cg backup  fetch-flow-cell -d -f HJFW5DSXX
Running command /bin/dsmc q archive */*HJFW5DSXX*.gpg
Found archived files for PDC query: */*HJFW5DSXX*.gpg
Encryption key found: /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.key.gpg
Flow cell found: /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg
```

**Comment by henrikstranneheim** [comment]
Very weird GitHub history and changes when merging to master. Restarting here: https://github.com/Clinical-Genomics/cg/pull/2517

**Pull request closed without merging**